### ISSUE TITLE
Unexpected command line interaction with pyproject.toml package

### ISSUE THREAD

**Issue by bluss** [astral-sh/uv#2722]
This command line interaction is surprising. My directory contains a pyproject.toml:

```
uv pip install --dry-run pyproject.toml  
Resolved 9 packages in 13ms
Would download 1 package
Would install 9 packages
 + attrs==23.2.0
 + jsonschema==4.21.1
 + jsonschema-specifications==2023.12.1
 + pyproject-toml==0.0.10
 + referencing==0.34.0
 + rpds-py==0.18.0
 + setuptools==69.2.0
 + toml==0.10.2
 + wheel==0.43.0
```

It has replaced `pyproject.toml` with package `pyproject-toml` from PyPI. (User error: must use `-r` to install from a pyproject.toml file.)
I'm sure this normalization sometimes makes sense, but here one mistake can lead to the wrong package being installed.

For the record, pip 24.0 behaves the same way.

```
uv -V
uv 0.1.26
platform: linux (x86_64)
```

**Comment by charliermarsh** [comment]
Lol oh no...

**Comment by zanieb** [comment]
We should save people from this :)

**Comment by charliermarsh** [comment]
Closed by #2746.

**Comment by bluss** [comment]
Thanks
